bot/connector-rest/src/main/kotlin/RestConnector.kt

```diff
@@ -205,4 +205,22 @@ class RestConnector(
                 else -> getTargetConnector(targetConnectorType)?.toConnectorMessage(message)?.invoke(this) ?: emptyList()
             }
         }
-}
+
+    private fun parseLocale(rawLocale: String): Locale {
```

Comment:
I think it's better to fix the problem at its source, from the caller.
Here, the code attempts a Locale.forLanguageTag(). The caller must pass this tag instead of what is currently being done, which is causing the problem, a Retrofit toString.

Look here: ConnectorRestClient#talk.

You will need to change the parameter from Local to string, then make the call as follows: 

`return getService(path).talk(locale.toLanguageTag(), query).execute()`

Reply:
After that, you should notice that the language display in the user interface is not correct. To fix this, use the normalizeLocaleCode function.